lib/ocpp/common/evse_security_impl.cpp

```diff
@@ -96,20 +96,30 @@ bool EvseSecurityImpl::is_ca_certificate_installed(const CaCertificateType& cert
     return this->evse_security->is_ca_certificate_installed(conversions::from_ocpp(certificate_type));
 }
 
-std::string EvseSecurityImpl::generate_certificate_signing_request(const CertificateSigningUseEnum& certificate_type,
-                                                                   const std::string& country,
-                                                                   const std::string& organization,
-                                                                   const std::string& common, bool use_tpm) {
-    return this->evse_security->generate_certificate_signing_request(conversions::from_ocpp(certificate_type), country,
-                                                                     organization, common, use_tpm);
+GetCertificateSignRequestResult
+EvseSecurityImpl::generate_certificate_signing_request(const CertificateSigningUseEnum& certificate_type,
+                                                       const std::string& country, const std::string& organization,
+                                                       const std::string& common, bool use_tpm) {
+    auto csr_response = this->evse_security->generate_certificate_signing_request(
+        conversions::from_ocpp(certificate_type), country, organization, common, use_tpm);
+
+    GetCertificateSignRequestResult result;
+    result.status = conversions::to_ocpp(csr_response.status);
+
+    if (csr_response.csr.has_value()) {
+        result.csr = csr_response.csr;
+    }
```

Comment:
This should do the job as well
```suggestion
    result.csr = csr_response.csr;
```